crowbar_framework/app/models/pacemaker_service.rb

```diff
@@ -369,6 +369,28 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     role.default_attributes["corosync"]["members"] = member_nodes.map{ |n| n.get_network_by_type("admin")["address"] }
     role.default_attributes["corosync"]["transport"] = role.default_attributes["pacemaker"]["corosync"]["transport"]
 
+    # Set up the second ring if dual_ring mode configured
```

Comment:
Please can you add this large chunk of code as a separate method?  Long methods really harm readability and maintainability.